### ISSUE TITLE
[BUG] Drawing a QNode decorated with `qml.cut_circuit_mc` fails

### ISSUE THREAD

**Issue by antalszava** [PennyLaneAI/pennylane#2494]
### Expected behavior

Draws:
```py
import pennylane as qml

dev = qml.device("default.qubit", wires=2, shots=1000)

@qml.cut_circuit_mc
@qml.qnode(dev)
def circuit(x):
    qml.RX(0.89, wires=0)
    qml.RY(0.5, wires=1)
    qml.RX(1.3, wires=2)

    qml.CNOT(wires=[0, 1])
    qml.WireCut(wires=1)
    qml.CNOT(wires=[1, 2])

    qml.RX(x, wires=0)
    qml.RY(0.7, wires=1)
    qml.RX(2.3, wires=2)
    return qml.sample(wires=[0, 2])

x = 0.3
qml.draw(circuit)(x)
```

### Actual behavior

The previous code fails.

### Additional information

A key part of the workflow is used by `cut_circuit_mc` is [this](https://github.com/PennyLaneAI/pennylane/blob/5d9da648260368ae11627c5bb093856d7948916d/pennylane/transforms/qcut.py#L1320) wrapper function upon executing the batch transform. For example, executing `circuit(x)` in the code above, `qnode_execution_wrapper_mc` is called as well as the contained [_wrapper](https://github.com/PennyLaneAI/pennylane/blob/5d9da648260368ae11627c5bb093856d7948916d/pennylane/transforms/qcut.py#L1334).
However, with `qml.draw(circuit)(x)`, the contained `_wrapper` function (which is key for setting the shots) is not called.

Also, `circuit.construct((x,), {})` fails too, which is crucial to get drawing and `qml.specs` to work with `cut_circuit_mc`.

### Tracebacks

```shell
TypeError                                 Traceback (most recent call last)
<ipython-input-29-917f99e3de7f> in <module>
     20 
     21 x = 0.3
---> 22 qml.draw(circuit)(x)

~/pennylane/pennylane/drawer/draw.py in wrapper(*args, **kwargs)
    187         try:
    188             qnode.expansion_strategy = expansion_strategy or original_expansion_strategy
--> 189             tapes = qnode.construct(args, kwargs)
    190         finally:
    191             qnode.expansion_strategy = original_expansion_strategy

~/pennylane/pennylane/transforms/batch_transform.py in _construct(args, kwargs)
    338             def _construct(args, kwargs):
    339                 qnode.construct(args, kwargs)
--> 340                 return self.construct(qnode.qtape, *targs, **tkwargs)
    341 
    342             wrapper.construct = _construct

~/pennylane/pennylane/transforms/batch_transform.py in construct(self, tape, *args, **kwargs)
    400             tape = self.expand_fn(tape, *args, **kwargs)
    401 
--> 402         tapes, processing_fn = self.transform_fn(tape, *args, **kwargs)
    403 
    404         if processing_fn is None:

~/pennylane/pennylane/transforms/qcut.py in cut_circuit_mc(tape, classical_processing_fn, auto_cutter, max_depth, shots, device_wires, **kwargs)
   1299     fragment_tapes = [remap_tape_wires(t, device_wires) for t in fragment_tapes]
   1300 
-> 1301     configurations, settings = expand_fragment_tapes_mc(
   1302         fragment_tapes, communication_graph, shots=shots
   1303     )

~/pennylane/pennylane/transforms/qcut.py in expand_fragment_tapes_mc(tapes, communication_graph, shots)
    795     """
    796     pairs = [e[-1] for e in communication_graph.edges.data("pair")]
--> 797     settings = np.random.choice(range(8), size=(len(pairs), shots), replace=True)
    798 
    799     meas_settings = {pair[0].id: setting for pair, setting in zip(pairs, settings)}

~/pennylane/pennylane/numpy/wrapper.py in _wrapped(*args, **kwargs)
    115 
    116         # evaluate the original object
--> 117         res = obj(*args, **kwargs)
    118 
    119         if isinstance(res, _np.ndarray):

~/autograd/autograd/tracer.py in f_wrapped(*args, **kwargs)
     46             return new_box(ans, trace, node)
     47         else:
---> 48             return f_raw(*args, **kwargs)
     49     f_wrapped.fun = f_raw
     50     f_wrapped._is_autograd_primitive = True

mtrand.pyx in numpy.random.mtrand.RandomState.choice()

~/anaconda3/lib/python3.8/site-packages/numpy/core/overrides.py in prod(*args, **kwargs)

~/anaconda3/lib/python3.8/site-packages/numpy/core/fromnumeric.py in prod(a, axis, dtype, out, keepdims, initial, where)
   3086     10
   3087     """
-> 3088     return _wrapreduction(a, np.multiply, 'prod', axis, dtype, out,
   3089                           keepdims=keepdims, initial=initial, where=where)
   3090 

~/anaconda3/lib/python3.8/site-packages/numpy/core/fromnumeric.py in _wrapreduction(obj, ufunc, method, axis, dtype, out, **kwargs)
     84                 return reduction(axis=axis, out=out, **passkwargs)
     85 
---> 86     return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
     87 
     88 

TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

### System information

```shell
Name: PennyLane
Version: 0.23.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /anaconda3/lib/python3.8/site-packages
Requires: semantic-version, autoray, autograd, retworkx, toml, networkx, scipy, cachetools, appdirs, numpy, pennylane-lightning
Required-by: PennyLane-Lightning, PennyLane-Orquestra, pennylane-qulacs, PennyLane-AQT, PennyLane-PQ, PennyLane-Forest, PennyLane-qsharp, PennyLane-SF, PennyLane-IonQ, PennyLane-Cirq, PennyLane-Honeywell, amazon-braket-pennylane-plugin, PennyLane-qiskit

Platform info:           Linux-5.13.0-39-generic-x86_64-with-glibc2.10
Python version:          3.8.5
Numpy version:           1.22.3
Scipy version:           1.8.0
Installed devices:
- default.gaussian (PennyLane-0.23.0)
- default.mixed (PennyLane-0.23.0)
- default.qubit (PennyLane-0.23.0)
- default.qubit.autograd (PennyLane-0.23.0)
- default.qubit.jax (PennyLane-0.23.0)
- default.qubit.tf (PennyLane-0.23.0)
- default.qubit.torch (PennyLane-0.23.0)
- lightning.qubit (PennyLane-Lightning-0.23.0)
- orquestra.forest (PennyLane-Orquestra-0.15.0)
- orquestra.ibmq (PennyLane-Orquestra-0.15.0)
- orquestra.qiskit (PennyLane-Orquestra-0.15.0)
- orquestra.qulacs (PennyLane-Orquestra-0.15.0)
- qulacs.simulator (pennylane-qulacs-0.17.0.dev0)
- aqt.noisy_sim (PennyLane-AQT-0.18.0)
- aqt.sim (PennyLane-AQT-0.18.0)
- projectq.classical (PennyLane-PQ-0.18.0.dev0)
- projectq.ibm (PennyLane-PQ-0.18.0.dev0)
- projectq.simulator (PennyLane-PQ-0.18.0.dev0)
- forest.numpy_wavefunction (PennyLane-Forest-0.18.0.dev0)
- forest.qvm (PennyLane-Forest-0.18.0.dev0)
- forest.wavefunction (PennyLane-Forest-0.18.0.dev0)
- microsoft.QuantumSimulator (PennyLane-qsharp-0.19.0)
- strawberryfields.fock (PennyLane-SF-0.21.0.dev0)
- strawberryfields.gaussian (PennyLane-SF-0.21.0.dev0)
- strawberryfields.gbs (PennyLane-SF-0.21.0.dev0)
- strawberryfields.remote (PennyLane-SF-0.21.0.dev0)
- strawberryfields.tf (PennyLane-SF-0.21.0.dev0)
- ionq.qpu (PennyLane-IonQ-0.21.0.dev0)
- ionq.simulator (PennyLane-IonQ-0.21.0.dev0)
- cirq.mixedsimulator (PennyLane-Cirq-0.22.0)
- cirq.pasqal (PennyLane-Cirq-0.22.0)
- cirq.qsim (PennyLane-Cirq-0.22.0)
- cirq.qsimh (PennyLane-Cirq-0.22.0)
- cirq.simulator (PennyLane-Cirq-0.22.0)
- honeywell.hqs (PennyLane-Honeywell-0.19.0.dev0)
- braket.aws.qubit (amazon-braket-pennylane-plugin-1.4.2.dev0)
- braket.local.qubit (amazon-braket-pennylane-plugin-1.4.2.dev0)
- qiskit.aer (PennyLane-qiskit-0.23.0.dev0)
- qiskit.basicaer (PennyLane-qiskit-0.23.0.dev0)
- qiskit.ibmq (PennyLane-qiskit-0.23.0.dev0)
- qiskit.ibmq.circuit_runner (PennyLane-qiskit-0.23.0.dev0)
- qiskit.ibmq.sampler (PennyLane-qiskit-0.23.0.dev0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by josh146** [comment]
@antalszava what happens when you draw `qml.cut_circuit`, does that also fail?

**Comment by antalszava** [comment]
No, it prints well :+1:  (I do have to change to expval because `qml.cut_circuit` doesn't support sampling).

**Comment by albi3ro** [comment]
This issue is solved by #4440
